### PR TITLE
[SILGen] Merge the two SILGen requests

### PR DESCRIPTION
**Pull request by hamishknight**
They both do essentially the same thing, with the single-file request being the single-element case of the whole-module request, with only the call to `getAllForModule` really caring about the difference.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test macOS

**Comment by hamishknight** [comment]
@swift-ci please test macOS

File: lib/SILGen/SILGen.cpp [hunk 1]
```diff
@@ -1817,6 +1817,8 @@ class SILGenModuleRAII {
 
 public:
   void emitSourceFile(SourceFile *sf) {
+    assert(sf->ASTStage == SourceFile::TypeChecked);
```
Comment (CodaFi) [inline lib/SILGen/SILGen.cpp:1820]: 👍 ✨

File: lib/SILGen/SILGen.cpp [hunk 2]
```diff
@@ -1835,7 +1837,7 @@ class SILGenModuleRAII {
     }
   }
 
-  SILGenModuleRAII(SILModule &M, ModuleDecl *SM) : SGM{M, SM} {}
+  SILGenModuleRAII(SILModule &M) : SGM{M, M.getSwiftModule()} {}
```
Comment (CodaFi) [inline lib/SILGen/SILGen.cpp:1840]: `explicit` here - to combat C++ being *bad*

File: lib/SILGen/SILGenRequests.cpp [hunk 3]
```diff
@@ -46,14 +46,18 @@ SourceLoc swift::extractNearestSourceLoc(const SILGenDescriptor &desc) {
...
 }
 
-evaluator::DependencySource SILGenSourceFileRequest::readDependencySource(
+evaluator::DependencySource SILGenerationRequest::readDependencySource(
     const evaluator::DependencyCollector &e) const {
   auto &desc = std::get<0>(getStorage());
+
+  // Whole module doesn't affect the dependency source.
+  if (auto *mod = desc.context.dyn_cast<ModuleDecl *>()) {
+    return {e.getActiveDependencySourceOrNull(), e.getActiveSourceScope()};
```
Comment (CodaFi) [inline lib/SILGen/SILGenRequests.cpp:55]: You can also explicitly return `{nullptr, active-scope}`, but I suppose this is more conservatively correct.

**Review by CodaFi — APPROVED** [review]
Only some nits. I wouldn't stop you from merging this.

File: lib/SILGen/SILGen.cpp (hunk 2 above)
Comment (hamishknight) [reply lib/SILGen/SILGen.cpp:1840]: Good catch!

File: lib/SILGen/SILGenRequests.cpp (hunk 3 above)
Comment (hamishknight) [reply lib/SILGen/SILGenRequests.cpp]: `nullptr` probably makes it a little more clear, I'll switch over to that for now.

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
@swift-ci please test Windows

**Pull request merged**